src/main/java/net/fabricmc/loader/metadata/ModDependencyImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.metadata;
+
+import java.util.List;
+
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.api.VersionParsingException;
+import net.fabricmc.loader.api.metadata.ModDependency;
+import net.fabricmc.loader.util.version.VersionPredicateParser;
+
+final class ModDependencyImpl implements ModDependency {
+	private final String modId;
+	private final List<String> matcherStringList;
+	private final String depAsStr;
+
+	ModDependencyImpl(String modId, List<String> matcherStringList) {
+		this.modId = modId;
+		this.matcherStringList = matcherStringList;
+
+		final StringBuilder builder = new StringBuilder("{");
+		builder.append(this.modId);
+		builder.append(" @ [");
+
+		for (int i = 0; i < matcherStringList.size(); i++) {
+			if (i > 0) {
+				builder.append(" || ");
+			}
+
+			builder.append(matcherStringList.get(i));
+		}
+
+		builder.append("]}");
+		this.depAsStr = builder.toString();
```

Comment:
Is toString being called during normal operation? If not, I'd move the string assembly to toString